src/GafferImageUI/WipeTool.cpp

```diff
@@ -0,0 +1,508 @@
+//////////////////////////////////////////////////////////////////////////
+
+class WipeTool::WipeHandle : public GafferUI::Gadget
+{
+
+	public :
+
+		WipeHandle()
+			:	Gadget(), m_pos( 0 ), m_dir( 1, 0 ), m_editable( true ), m_dragHandle( HandleSelect::None )
+		{
+			mouseMoveSignal().connect( boost::bind( &WipeHandle::mouseMove, this, ::_2 ) );
+			buttonPressSignal().connect( boost::bind( &WipeHandle::buttonPress, this, ::_2 ) );
+			buttonReleaseSignal().connect( boost::bind( &WipeHandle::buttonRelease, this, ::_2 ) );
+			dragBeginSignal().connect( boost::bind( &WipeHandle::dragBegin, this, ::_1, ::_2 ) );
+			dragEnterSignal().connect( boost::bind( &WipeHandle::dragEnter, this, ::_1, ::_2 ) );
+			dragMoveSignal().connect( boost::bind( &WipeHandle::dragMove, this, ::_2 ) );
+			dragEndSignal().connect( boost::bind( &WipeHandle::dragEnd, this, ::_2 ) );
+			leaveSignal().connect( boost::bind( &WipeHandle::leave, this ) );
+		}
+
+		Imath::Box3f bound() const override
+		{
+			// The wipe indicator is an infinite line, so we don't have a sensible bound
+			return Box3f();
+		}
+
+		void setPosition( const Imath::V2f &p )
+		{
+			setWipeInternal( p, m_dir );
+		}
+
+		const Imath::V2f &getPosition() const
+		{
+			return m_pos;
+		}
+
+		void setDirection( const Imath::V2f &d )
+		{
+			if( d != V2f( 0 ) )
+			{
+				setWipeInternal( m_pos, d.normalized() );
+			}
+		}
+
+		const Imath::V2f &getDirection() const
+		{
+			return m_dir;
+		}
+
+		void setEditable( bool editable )
+		{
+			m_editable = editable;
+		}
+
+		bool getEditable() const
+		{
+			return m_editable;
+		}
+
+	protected :
+
+		void renderLayer( Layer layer, const Style *style, RenderReason reason ) const override
+		{
+			if( layer != Layer::Front )
+			{
+				return;
+			}
+
+			float rsf = rasterScaleFactor();
+			float handleRadius = 30.0f * rsf;
+
+			Color4f bright( 0.8f, 0.8f, 0.8f, 0.5f );
+			Color4f dark( 0.2f, 0.2f, 0.2f, 0.5f );
+
+			if( isSelectionRender( reason ) )
+			{
+				if( m_editable )
+				{
+					// Draw one thick handle for selection
+					renderHandle( style, handleRadius, rsf * g_handleThickness, 0.0f, dark );
+				}
+			}
+			else
+			{
+				// Draw a thin bright handle offset from a dark handle, so that it will
+				// be visible on any background
+				renderHandle( style, handleRadius, rsf * 1.7f, -rsf * 0.5, dark );
+				renderHandle( style, handleRadius, rsf * 1.7f, rsf * 0.5, bright );
+			}
+		}
+
+		unsigned layerMask() const override
+		{
+			return (unsigned)Layer::Front;
+		}
+
+		Imath::Box3f renderBound() const override
+		{
+			Box3f b;
+			b.makeInfinite();
+			return b;
+		}
+
+	private :
+
+		void renderHandle( const Style *style, float rotateHandleSize, float thickness, float offset, const Color4f &color ) const
+		{
+			// Calculate a start and end point for the section of the infinite line defined by m_pos
+			// and m_dir that passes through the viewport.  Little bit complicated, but makes everything
+			// else simple.
```

Comment:
My first pass was a very naive version of exactly that, which ran into weird numerical precision issues if the rotate handle was really far from the origin, at which point I just went for the fully complicated and maximally robust version.  But after implementing that, I realized that there was a simpler option that would work fine - do take the viewport into account, but only to find the point on the line which is closest to the viewport center, and base the drawing on that.  That is robust, and moderately simpler - since the complexity bothered you, I went ahead and implemented that version.